src/lib/y2storage/encryption_processes/fde_tools.rb

```diff
@@ -0,0 +1,104 @@
+# Copyright (c) [2023] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast2/execute"
+require "yast2/systemd/service"
+require "y2storage/secret_attributes"
+
+module Y2Storage
+  module EncryptionProcesses
+    # Auxiliary class to interact with the utilities provided by the fde-tools package
+    class FdeTools
+      include SecretAttributes
+      include Yast::Logger
+
+      # Location of the fdectl command
+      FDECTL = "/usr/sbin/fdectl".freeze
+      private_constant :FDECTL
+
+      # Passphrase that was used to encrypt the device and that is needed to
+      # perform several of the steps
+      #
+      # @return [String]
+      secret_attr :recovery_password
+
+      # Constructor
+      def initialize(recovery_password = nil)
+        self.recovery_password = recovery_password
+      end
+
+      # Whether fde-tools detect a working TPM2 chip in the system
+      #
+      # @return [Boolean]
+      def tpm_present
+        Yast::Execute.on_target!(FDECTL, "tpm-present")
+        log.info "FDE: TPMv2 detected"
+        true
+      rescue Cheetah::ExecutionFailed
+        log.info "FDE: TPMv2 not detected"
+        false
+      end
+
+      alias_method :tpm_present?, :tpm_present
```

Comment:
NP: why not to directly call it `#tpm_present?`

Reply:
No strong reason. I just wanted to have the method names equivalent to the `fdectl` commands (obviously turning hyphens into underscores). Since the `fdectl` subcommands are `tpm-present`, `add-secondary-password` and `add-secondary-key`...